File: playful/app/models/file_type.rb
class FileType < ActiveRecord::Base
  has_many :base_files

  validates_uniqueness_of :name
  validates_presence_of :name, :subtype, :extension, :ruby_class, :mime_type, :scan_type
  validates_format_of :mime_type, :with => /^[^\/]+\/[^\/]+(\s*,\s*[^\/]+\/[^\/]+)*?$/, :multiline => true

  AUDIO = 'Audio'
  IMAGE = 'Image'
  VIDEO = 'Video'
end

File: playful/app/assets/javascripts/app/models/files/filetype.js
App.FileType = DS.Model.extend({
    baseFiles: DS.hasMany('baseFile'),
    name: DS.attr('string'),
    subtype: DS.attr('string'),
    extension: DS.attr('string'),
    mimeType: DS.attr('string'),
    scanType: DS.attr('string'),
    createdAt: DS.attr('date'),
    updatedAt: DS.attr('date')
});
